nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,315 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) mkIf types;
+  inherit (lib.lists) optionals;
+  inherit (lib.attrsets) optionalAttrs;
+  inherit (lib.strings) concatStringsSep escapeShellArg;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+
+  cfg = config.services.pr-tracker;
+in
+{
+  options.services.pr-tracker = {
+    enable = mkEnableOption "pr-tracker";
+
+    package = mkPackageOption pkgs "pr-tracker" { };
+
+    githubApiTokenFile = mkOption {
+      type = types.path;
+      example = "/run/secrets/gh-token";
+
+      description = ''
+        Path to a file containing your GitHub API token like so:
+
+        ```env
+        ghp_...
+        ```
+
+        ::: {.note}
+        The contents of this file will be the stdin of pr-tracker.
+        :::
+      '';
+    };
+
+    nixpkgsClone = {
+      cloneDir = mkOption {
+        type = types.path;
+        example = "/home/nixos/git/nixpkgs";
+
+        default = "${cfg.dataDir}/nixpkgs";
+        defaultText = "/var/lib/pr-tracker/nixpkgs";
+        description = ''
+          The path to the cloned nixpkgs pr-tracker will use.
+
+          ::: {.note}
+          If left as the default value this repo will automatically be cloned before
+          the pr-tracker server starts, otherwise you are responsible for ensuring
+          the directory exists with appropriate ownership and permissions.
+          :::
+        '';
+      };
+
+      remote = mkOption {
+        type = types.str;
+        example = "upstream";
+
+        default = "origin";
+        description = ''
+          The remote name in the repository corresponding to upstream Nixpkgs.
+        '';
+      };
+
+      managedByModule = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether you want this service to manage a clone of the nixpkgs
+          repo in `cfg.dataDir`.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        example = "15min";
+
+        default = "30min";
+        description = ''
+          How often to fetch nixpkgs if `cfg.nixpkgsClone.managedByModule` is true.
+        '';
+      };
+    };
+
+    userAgent = mkOption {
+      type = types.str;
+      example = "my pr-tracker";
+
+      description = ''
+        The User-Agent string to use when contacting the GitHub API.
+      '';
+    };
+
+    sourceUrl = mkOption {
+      type = types.str;
+      example = "https://github.com/me/my-pr-tracker-fork";
+
+      default = "https://git.qyliss.net/pr-tracker";
+      description = ''
+        The URL where users can download the program's source code.
+      '';
+    };
```

Comment:
I think we should use cfg.package.src here instead

Reply:
or cfg.package.homepage

> I'd also rather it didn't have a default

That would be very bad UX for everyone trying to just host the default package to get some load of the hosted instance. I don't know any software from the top of my head which does that, usually the URL is just hardcoded.

Reply:
I like the idea of using `cfg.package.meta.homepage`

What do you think @alyssais ?

Reply:
Again, this is too easy to forget if somebody makes a change. It would be fine for that to be an example, but there's no default for a reason.

Reply:
It also wouldn't tell you which specific version of the code the nistance was running.

If you do end up with optional nginx integration though, you could automate compliance through that, by having it automatically serve a source tarball and set that URL here. That way it would always reflect the actual source code being used, even if changes were made.  Here's an [example](https://git.qyliss.net/nixlib/tree/modules/server/mail/public-inbox/default.nix?id=3a58f35ee8ee6da08014cded4ee53dd8349e33b4#n6) of making such a tarball.